app/src/main/java/nu/yona/app/security/SecurityKey.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * <?xml version="1.0" encoding="utf-8"?><!--
+ * ~ Copyright (c) 2018 Stichting Yona Foundation
+ *   ~
+ *   ~ This Source Code Form is subject to the terms of the Mozilla Public
+ *   ~ License, v. 2.0. If a copy of the MPL was not distributed with this
+ *   ~ file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ *   -->
+ */
+
+package nu.yona.app.security;
+
+
+import android.os.Build;
+import android.util.Base64;
+
+import java.security.GeneralSecurityException;
+import java.security.KeyPair;
+
+import javax.crypto.Cipher;
+import javax.crypto.SecretKey;
+import javax.crypto.spec.GCMParameterSpec;
+import javax.crypto.spec.IvParameterSpec;
+
+import nu.yona.app.utils.AppUtils;
+
+class SecurityKey
+{
+	private static final String RSA_MODE = "RSA/ECB/PKCS1Padding";
+	private static final String AES_MODE_FOR_POST_API_23 = "AES/GCM/NoPadding";
+	private static final String AES_MODE_FOR_PRE_API_18 = "AES/CBC/PKCS5Padding";
+
+	private SecretKey secretKey;
+	private KeyPair keyPair;
+
+	SecurityKey(SecretKey secretKey)
+	{
+		this.secretKey = secretKey;
+	}
+
+	SecurityKey(KeyPair keyPair)
+	{
+		this.keyPair = keyPair;
+	}
+
+	String encrypt(String token)
+	{
+		if (token == null)
+		{
+			return null;
+		}
+		try
+		{
+			Cipher cipher = getCipher(Cipher.ENCRYPT_MODE);
+
+			byte[] encrypted = cipher.doFinal(token.getBytes());
+			return Base64.encodeToString(encrypted, Base64.URL_SAFE);
+		}
+		catch (GeneralSecurityException e)
+		{
+			AppUtils.reportException(SecurityKey.class, e, Thread.currentThread());
+		}
+		//Unable to encrypt Token
+		return null;
```

Comment:
So, fail with an exception rather than returning null. This is particularly wrong here, as ``null`` is a valid return value for this method.

Reply:
Done.